### ISSUE TITLE
Calling pm.collectionVariables.set in a pre-request script deletes all disabled collection level variables

### ISSUE THREAD

**Issue by shaneocall** [a85/POSTMan-Chrome-Extension#10709]
### Is there an existing issue for this?

- [X] I have searched the tracker for existing similar issues

### Describe the Issue

If I have variables in the collection which are disabled (checkbox is unticked). Whenever a script calls `pm.collectionVariables.set("myKey", "myValue")` then the disabled collection variables are deleted.

This is a regression, this did not happnen with previous versions of Postman

### Steps To Reproduce

1. Create a collection with one variable enabled (`myVar1`), one disabled (`myVar2`) and one which will be set from a script (`mySetVarDate`) like below: 
![image](https://user-images.githubusercontent.com/16311180/156265080-4323bfc4-2390-4a46-9900-daa11f97450b.png)

2. Create a `GET` request with a `pre-request` script like below
![image](https://user-images.githubusercontent.com/16311180/156265148-8e1de7f5-da05-414e-9127-0fb67ee505fc.png)

3. Execute the request
4. Go back and check the collection variables, `myVar2` has disappeared:

![image](https://user-images.githubusercontent.com/16311180/156265364-9f12c8b9-2011-48d7-b086-e4edcaa96ff9.png)

### Environment Information

```markdown
- Operating System: Windows Server 2019 Datacenter
- Platform Type: Native App
- Postman Version: 
Postman for Windows
Version 9.14.7
UI Version 9.14.7-ui-220228-1230
Desktop Platform Version 9.14.0
Architecture x64

OS Platform win32 10.0.17763
```

**Comment by abhijeetborole** [comment]
Hey @shaneocall, thanks for reporting this issue. We identified and fixed it on `v9.14.16`. Could you please check it out and verify that the issue is fixed for you on `v9.14.16`? Do let us know if you face any other issues!

**Comment by DannyDainton** [comment]
Closing this as the issue has been fixed in the latest version.

**Comment by shaneocall** [comment]
> Hey @shaneocall, thanks for reporting this issue. We identified and fixed it on `v9.14.16`. Could you please check it out and verify that the issue is fixed for you on `v9.14.16`? Do let us know if you face any other issues!

Confirmed fixed in version  9.15.2, thanks!
